cmd/spire-agent/cli/common/defaults_posix.go

```diff
@@ -2,9 +2,24 @@
 
 package common
 
+import (
+	"os"
+)
+
 const (
-	// DefaultSocketPath is the SPIRE agent's default socket path
-	DefaultSocketPath = "/tmp/spire-agent/public/api.sock"
+	// DefaultRunSocketPath is the SPIRE agent's default socket path
+	DefaultRunSocketPath = "/tmp/spire-agent/public/api.sock"
 	// DefaultAdminSocketPath is the SPIRE agent's default admin socket path
 	DefaultAdminSocketPath = "/tmp/spire-agent/private/admin.sock"
 )
+
+// DefaultSocketPath is the SPIRE agent's default socket path
+var DefaultSocketPath string
```

Comment:
So, it is going to be used for `api` calls and `healthchecks` except for run command where we are going to keep using 'DefaultRunSocketPath' 
Is it expected to change healthchecks?
And if not may we the code to change default inside `AddOSFlags` instead?

Reply:
We chatted about this. I believe the current plan is to do everything but run in one pr, and run in a future pr.